nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/molecule/CanvasMolecule.java

```diff
@@ -333,52 +325,96 @@ String getHit() {
-                    molecule.clearSelected();
-                }
-                return true;
+                selectMolecule(selectMode);
             } else {
-                System.out.println(getHit());
-                String aName = getHit();
-                Molecule molecule = Molecule.get(molName);
-                try {
-                    molecule.selectAtoms(aName);
-                } catch (InvalidMoleculeException ex) {
-                    log.warn(ex.getMessage(), ex);
-                }
-
-                return true;
+                selectAtom(selectMode);
             }
+            return true;
         }
     }
 
+    /**
+     * Moves the molecule around the canvas. If a handle is selected, the handle can be
+     * moved to adjust the size of the molecule but, it cannot be moved past another handle.
+     * (i.e. The molecule cannot be sized to a negative width or height)
+     * @param bounds The bounds of the canvas.
+     * @param world The bounds of the canvas in the units of the canvas axis.
+     * @param start The starting position.
+     * @param pos The new position.
+     */
     @Override
     public void move(double[][] bounds, double[][] world, double[] start, double[] pos) {
         double dx = pos[0] - start[0];
         double dy = pos[1] - start[1];
-        bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
-        bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
-        by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
-        by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+        if (activeHandle < 0) {
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+
+        } else if (activeHandle == 0) { // upper left
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+            bx1 = Math.min(bx1, bx2);
+            by1 = Math.min(by1, by2);
+        } else if (activeHandle == 1) { // upper right
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+            bx2 = Math.max(bx1, bx2);
+            by1 = Math.min(by1, by2);
+        } else if (activeHandle == 2) { // bottom right
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+            bx2 = Math.max(bx1, bx2);
+            by2 = Math.max(by1, by2);
+        } else if (activeHandle == 3) { // bottom left
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+            bx1 = Math.min(bx1, bx2);
+            by2 = Math.max(by1, by2);
+        }
```

Comment:
@brucejohnson when adjusting the molecules or the journal text, the handles no longer go past each other...but if you bring the handles right together, it can be really hard/impossible to reselect the object since there is no space, not sure if we should only allow the handles to get no less than x distance away from each other? and if so what should that distance be per position type?

Reply:
Yes, I think we should limit this.  A reasonable minimum might be twice the handle size

Reply:
Added a limit of twice the handle size